var/spack/repos/builtin/packages/py-fparser/package.py

```diff
@@ -0,0 +1,65 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class PyFparser(PythonPackage):
+    """Parser for Fortran 77..2003 code."""
+
+    homepage = "https://github.com/stfc/fparser"
+    url      = "https://github.com/stfc/fparser/archive/0.0.5.tar.gz"
+    giturl   = "https://github.com/stfc/fparser.git"
+
+    version('0.0.6', git=giturl,
+            commit='638c51ec57cf17624505b70321c3784e356b8910')
+    version('0.0.5', git=giturl,
+            commit='a3ff86b635f7bd7bd281ee94cbd3d9455b288fd9')
+    version('develop', git=giturl, branch='master')
+
+    depends_on('py-setuptools', type='build')
+
+    depends_on('python', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+
+    # Use type='test' when available:
```

Comment:
This is now available! See #5132.

Reply:
Hi @adamjstewart , my tests seem to require the `type='build'` deptype for py-pytest. If I try `type='test'`, as in the latest commit (3a873df), the tests fail with the following error messages:

```
running install_scripts
==> RUN-TESTS: install-time tests [import_module_test]
==> 'py.test'
Traceback (most recent call last):
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-pytest-3.0.7-hlae3ijeej6rl5whyakmllvzss4urvp4/bin/py.test", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-setuptools-35.0.2-ddwvbav2efm7kfig5kdj4eigvdsudmfz/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3036, in <module>
    @_call_aside
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-setuptools-35.0.2-ddwvbav2efm7kfig5kdj4eigvdsudmfz/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3020, in _call_aside
    f(*args, **kwargs)
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-setuptools-35.0.2-ddwvbav2efm7kfig5kdj4eigvdsudmfz/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3049, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-setuptools-35.0.2-ddwvbav2efm7kfig5kdj4eigvdsudmfz/lib/python2.7/site-packages/pkg_resources/__init__.py", line 654, in _build_master
    ws.require(__requires__)
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-setuptools-35.0.2-ddwvbav2efm7kfig5kdj4eigvdsudmfz/lib/python2.7/site-packages/pkg_resources/__init__.py", line 968, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/mwoods/spack/opt/spack/linux-rhel7-x86_64/gcc-7.2.0/py-setuptools-35.0.2-ddwvbav2efm7kfig5kdj4eigvdsudmfz/lib/python2.7/site-packages/pkg_resources/__init__.py", line 854, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pytest==3.0.7' distribution was not found and is required by the application
```

Reply:
@scheibelp Can you look into this?